p2p/messenger.go

```diff
@@ -56,6 +58,13 @@ func (w *bufferedDelimitedWriter) Flush() error {
 	return w.Writer.Flush()
 }
 
+// messenger is an interface to wrap p2p messaging implementation
+type messenger interface {
```

Comment:
I understand that this looks weird, but go convention is to define the interface where its used, unless its for packages that define multiple interfaces. So I would put this back to where it was before. i.e. the library would not care about defining an interface for it to be used with.